app/src/main/java/com/example/tasky/common/domain/TokenManager.kt

```diff
@@ -0,0 +1,46 @@
+package com.example.tasky.common.domain
+
+import android.util.Log
+import com.example.tasky.common.domain.repository.TokenRemoteRepository
+import kotlinx.coroutines.flow.first
+import java.time.ZonedDateTime
+import javax.inject.Inject
+
+class TokenManager @Inject constructor(
+    private val tokenRepository: TokenRemoteRepository,
+    private val sessionStateManager: SessionStateManager
+) {
+    companion object {
+        private val TAG = "TokenManager"
+    }
+
+    suspend fun shouldRefreshToken() {
+        val now = ZonedDateTime.now()
+        val currentTimeInMilliseconds = now.toInstant().toEpochMilli()
+        val expirationTime = sessionStateManager.getAccessTokenExpiration().first() ?: 0L
+
+        if (currentTimeInMilliseconds > expirationTime) {
+            refreshToken()
+        }
+    }
```

Comment:
Alternatively, you can just make this return `currentTimeInMilliseconds > expirationTime` and then have something like this:
```
if(shouldRefreshToken()) {
    refreshToken()
}
```
Is just as readable as what I shared above